woocommerce/wc-admin
# Add dashboard chart interval preferences

Fixes #1186

Adds an interval selector to dashboard charts and saves them to user preferences to be persisted when accessing the dashboard in the future.

### Screenshots
<img width="766" alt="screen shot 2019-01-03 at 3 52 10 pm" src="https://user-images.githubusercontent.com/10561050/50627675-93fb0980-0f6f-11e9-86fd-d06fd201ec7b.png">

### Detailed test instructions:

1.  Visit the Dashboard.
2.  Make sure you've select a date range or preset long enough for multiple intervals.
3.  Check that the interval dropdown adjusts based on the date range selected (try yesterday or today to limit to only hourly).
4.  Change the interval to another type and check that the chart updates accordingly.
5.  Check that on refresh the chart interval is persisted.

### Questions
* The persisted interval isn't immediately loaded resulting in a secondary load and flash as the charts refresh.  Do we want to delay charts from showing while user preferences are being requested or is it okay as is?
* The interval selector positioning on mobile is a bit hacky with position absolute.  This should not cause issues with overlapping, but would creating another prop for `SectionHeader` be a better idea so we can restructure those elements?  I'm concerned that adds extra surface area to a component for a single use case.
